### PR TITLE
chore: add eslint with antfu's eslint config

### PR DESCRIPTION
**Pull request by bdbch**
Added eslint with antfu's config and also added the defaultFormatter for Code plus the extension recommendation.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "editor.defaultFormatter": "dbaeumer.vscode-eslint"
```
Comment (anbraten) [inline .vscode/settings.json:2]: Some additional settings taken from the nimbus code base that might be helpful:

```suggestion
  "editor.defaultFormatter": "dbaeumer.vscode-eslint",
  
   // Disable the default formatter, use eslint instead
  "prettier.enable": false,
  "editor.formatOnSave": false,

  // Auto fix
  "editor.codeActionsOnSave": {
    "source.fixAll": "explicit",
    "source.organizeImports": "never"
  },
  
  // Silent the stylistic rules in you IDE, but still auto fix them
  "eslint.rules.customizations": [
    { "rule": "style/*", "severity": "off" },
    { "rule": "*-indent", "severity": "off" },
    { "rule": "*-spacing", "severity": "off" },
    { "rule": "*-spaces", "severity": "off" },
    { "rule": "*-order", "severity": "off" },
    { "rule": "*-dangle", "severity": "off" },
    { "rule": "*-newline", "severity": "off" },
    { "rule": "*quotes", "severity": "off" },
    { "rule": "*semi", "severity": "off" }
  ],

  // Enable eslint for all supported languages
  "eslint.validate": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact",
    "vue",
    "html",
    "markdown",
    "json",
    "jsonc",
    "yaml"
  ]
```

File: packages/core/package.json [hunk 2]
```diff
@@ -2,9 +2,23 @@
   "name": "tsky",
   "version": "0.1.0",
   "license": "MIT",
+  "scripts": {
+    "lint": "eslint ./src",
+    "lint:fix": "eslint ./src --fix"
+  },
+  "dependencies": {
+    "@antfu/eslint-config": "^3.11.2"
```
Comment (anbraten) [inline packages/core/package.json:10]: Should probably be a `devDependencies` as well 😉

**Review by anbraten — APPROVED** [review]
Just some conflict and a nit, rest lgtm. Can't wait to have some auto formatting 😊

File: packages/core/package.json (hunk 2 above)
Comment (bdbch) [reply packages/core/package.json]: 👍 updated!

**Comment by bdbch** [comment]
updated my PR - should be good to merge now.

**Pull request merged**